src/common-utils/main.sh

```diff
@@ -567,6 +568,53 @@ if [ "${INSTALL_ZSH}" = "true" ]; then
     fi
 fi
 
+# *********************************
+# ** Enable shell history **
+# *********************************
+
+if [ "${ALLOW_SHELL_HISTORY}" = "true" ]; then
+    echo "Activating feature 'shell-history'"
+    echo "User: ${USERNAME}     User home: ${user_home}"
+
+    if ! command -v uuidgen &> /dev/null; then
+        sudo apt-get update
+        sudo apt-get install -y uuid-runtime
+    fi
+    # Create the shell history directory in the mounted volume
+    DEVCONTAINER_ID=$(uuidgen)
+    HISTORY_DIR="/devcontainers/${DEVCONTAINER_ID}/shellHistory"
+    USER_HISTORY_FILE="${user_home}/.bash_history"
+    VOLUME_HISTORY_FILE="${HISTORY_DIR}/.bash_history"
+
+    # Create the history directory in the volume, if it doesn’t already exist
+    sudo mkdir -p "${HISTORY_DIR}"
+    sudo chown -R "${USERNAME}" "${HISTORY_DIR}"
+    sudo chmod -R u+rwx "${HISTORY_DIR}"
+
+    # Ensure the volume's history file exists and set permissions
+    if [[ ! -f "${VOLUME_HISTORY_FILE}" ]]; then
+        # Create an empty history file if it doesn’t already exist
+        sudo touch "${VOLUME_HISTORY_FILE}"
+        sudo chown -R "${USERNAME}" "${VOLUME_HISTORY_FILE}"
+        sudo chmod -R u+rwx "${VOLUME_HISTORY_FILE}"
+    fi
+
+    # Create or update the user’s .bash_history to append to the volume’s history
+    if [[ ! -f "${USER_HISTORY_FILE}" ]]; then
+        sudo touch "${USER_HISTORY_FILE}"
```

Comment:
This will be overwritten by the symbolic link below. I think this can be removed.

Reply:
ok